test/saithrift/tests/saiacl.py

```diff
@@ -1634,3 +1634,273 @@ def runTest(self):
+                                                out_port,
+                                                src_l4_port,
+                                                dst_l4_port)
+            acl_entry_id = sai_thrift_create_acl_entry(
+                                                self.client,
+                                                acl_table_id,
+                                                entry_priority,
+                                                action,
+                                                addr_family,
+                                                mac_src,
+                                                mac_src_mask,
+                                                mac_dst,
+                                                mac_dst_mask,
+                                                ip_src,
+                                                ip_src_mask,
+                                                ip_dst,
+                                                ip_dst_mask,
+                                                ip_proto,
+                                                in_ports,
+                                                out_ports,
+                                                in_port,
+                                                out_port,
+                                                src_l4_port,
+                                                dst_l4_port,
+                                                ingress_mirror_id,
+                                                egress_mirror_id)
+
+            # bind this ACL table to VLAN object id
+            print "Binding ACL to VLAN"
+            attr_value = sai_thrift_attribute_value_t(oid=acl_table_id)
+            attr = sai_thrift_attribute_t(id=SAI_VLAN_ATTR_INGRESS_ACL, value=attr_value)
+            self.client.sai_thrift_set_vlan_attribute(vlan_oid, attr)
+
+            print "Asserting ACL table entry"
+            assert acl_table_id > 0, 'acl_entry_id is <= 0'
+            assert acl_entry_id > 0, 'acl_entry_id is <= 0'
+            print '#### ACL \'DROP, src 12.12.12.1/255.255.255.0, in_ports[ptf_intf_1]\' Applied ####'
+            print '#### Sending      ', router_mac, '| 00:11:22:33:44:56 | 192.168.0.1 | 12.12.12.1 | @ ptf_intf 2'
+            # send the same packet
+            send_packet(self, 1, str(pkt))
+            # ensure packet is dropped
+            # check for absence of packet here!
+            print '#### NOT Expecting 00:11:22:33:44:55 |', router_mac, '| 12.12.12.1 | 192.168.0.1 | @ ptf_intf 1'
+            verify_no_packet(self, exp_pkt, 0)
+            print "ACL blocked packet as expected"
+
+            print "Sending packet port 3 -> port 1"
+            send_packet(self, 2, str(pkt))
+            print "Verifying packet at port 1"
+            verify_no_packet(self, exp_pkt, 0)
+
+
+            # ===== L3AclTableTest_III verification =======
```

Comment:
Need to unbind ACL with vlan first, otherwise the packet will still be dropped

            attr_value = sai_thrift_attribute_value_t(oid=SAI_NULL_OBJECT_ID)
            attr = sai_thrift_attribute_t(id=SAI_VLAN_ATTR_INGRESS_ACL, value=attr_value)
            self.client.sai_thrift_set_vlan_attribute(vlan_oid, attr)

Reply:
The  Test Description says to dynamically  change the bind point type to vlan and port . So didn't unbind the acl from the vlan  and applied it dynamically to the Port in the vlan 
https://github.com/opencomputeproject/SAI/blob/master/test/saithrift/tests/SAITest.md
```
After “L3AclTableTest – II”, bind the same ACL table to a port. Test is to verify dynamically changing the bind point type of an ACL table from VLAN to a PORT. 
```

Reply:
Changing the binding from vlan to port means unbinding from the vlan, and binding to the port
If you just add binding to the port, the table remains bounded to the vlan, and also bounded to the port
While test is about removing the vlan binding, and binding to the port instead